tests/integration/test_arithmetic.py

```diff
@@ -230,3 +230,21 @@ def run_rem() -> int:
     run_int_fn(hugr, expected=-42, fn_name="run_neg")
     run_int_fn(hugr, expected=-2, fn_name="run_div")
     run_int_fn(hugr, expected=2, fn_name="run_rem")
+
+
+def test_angle_exec(validate, run_float_fn):
+    module = GuppyModule("test_angle_exec")
+    module.load_all(angles)
+
+    @guppy(module)
+    def main() -> float:
+        a1 = pi
+        a2 = pi * 2
+        a3 = -a1 + a2 * -3
+        a3 -= a1
+        a3 += 2 * a1
+        return float(a3)
+    hugr = module.compile()
+    validate(hugr)
+    import math
+    run_float_fn(hugr, expected=-6 * math.pi)
```

Comment:
Hmmmm, the worrying thing here is that `expected==`. We appear to be performing exact equality on floats, and where the expected value uses all the precision, to boot....

I think `pytest` has `assert_close`, maybe `run_float_fn` should just return the `float` (even if we keep `run_int_fn` as is to avoid a major refactor)

Reply:
Added `pytest.approx` here